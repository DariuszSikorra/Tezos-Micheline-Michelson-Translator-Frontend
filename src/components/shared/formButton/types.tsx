export type ButtonProps = {
  type: 'submit' | 'reset' | 'button';
  label: string;
  className?: string;
  onClick?: any;
  disabled?: boolean;
  stylingType?: string;
};
