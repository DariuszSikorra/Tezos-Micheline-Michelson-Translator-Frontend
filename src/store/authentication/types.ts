export const AUTHENTICATION_SET_AUTH = 'AUTHENTICATION_SET_AUTH';
export const AUTHENTICATION_LOGIN = 'AUTHENTICATION_LOGIN';
export const AUTHENTICATION_SUCCESS = 'AUTHENTICATION_SUCCESS';
export const AUTHENTICATION_FAIL = 'AUTHENTICATION_FAIL';


export type authCredentials = {
  username: string,
  password: string,
};

export type AuthenticationState = {
  token: string;
  isLogged: string;
};

export interface IAuthenticationLogin {
  type: typeof AUTHENTICATION_LOGIN,
  payload: authCredentials,
};
export interface IAuthenticationSuccess {
  type: typeof AUTHENTICATION_SUCCESS,
  token: string,
};
export interface IAuthenticationFail {
  type: typeof AUTHENTICATION_FAIL,
  error: string,
  status: string
};

export type AuthActionTypes =
  | IAuthenticationLogin
  | IAuthenticationSuccess
  | IAuthenticationFail;