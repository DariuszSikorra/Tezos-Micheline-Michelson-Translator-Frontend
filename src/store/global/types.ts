import { LibraryState } from "../library/types";
import { TranslatorState, TranslatorMessageState } from "../translator/types";
import { MessageState } from "../message/types";
import { AuthenticationState } from "../authentication/types";

export interface IState {
  library: LibraryState;
  message: MessageState;
  auth: AuthenticationState;
  translator: TranslatorState;
  translatorMessage: TranslatorMessageState;
  pending: boolean;
  error?: string;
}