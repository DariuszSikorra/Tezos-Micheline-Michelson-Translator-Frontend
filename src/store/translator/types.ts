export const TRANSLATOR_SET_MODE = 'TRANSLATOR_SET_MODE';
export const TRANSLATOR_FETCH_MICHELSON_TO_MICHELINE = 'TRANSLATOR_FETCH_MICHELSON_TO_MICHELINE';
export const TRANSLATOR_FETCH_MICHELINE_TO_MICHELSON = 'TRANSLATOR_FETCH_MICHELINE_TO_MICHELSON';
export const TRANSLATOR_SET_MICHELINE = 'TRANSLATOR_SET_MICHELINE';
export const TRANSLATOR_SET_MICHELSON = 'TRANSLATOR_SET_MICHELSON';
export const TRANSLATOR_SET_ERROR = 'TRANSLATOR_SE_ERROR';
export const TRANSLATOR_FLUSH_TRANSLATION = 'TRANSLATOR_FLUSH_TRANSLATION';

export type TranslatorState = {
  mode: Modes;
  michelson: string;
  micheline: string;
  error: string;
};

export type TranslatorAction = {
  type: string;
  mode: Modes;
  translation: string;
  error?: any;
};

export enum Modes {
  MICHELINEMICHELSON = 'michelinemichelson',
  MICHELSONMICHELINE = 'michelsonmicheline',
}

export interface IState {
  pending: boolean;
  translator: TranslatorState;
  error?: any;
}

export interface ITranslatorSetMode {
  type: typeof TRANSLATOR_SET_MODE;
  mode: Modes;
}

export interface ITranslatorFetchMichelineToMichelson {
  type: typeof TRANSLATOR_FETCH_MICHELINE_TO_MICHELSON;
  payload: string;
}

export interface ITranslatorFetchMichelsonToMicheline {
  type: typeof TRANSLATOR_FETCH_MICHELSON_TO_MICHELINE;
  payload: string;
}

export interface ITranslatorSetMichelson {
  type: typeof TRANSLATOR_SET_MICHELSON;
  status: number;
  translation: string;
}

export interface ITranslatorSetMicheline {
  type: typeof TRANSLATOR_SET_MICHELINE;
  status: number;
  translation: string;
}

export interface ITranslatorSetError {
  type: typeof TRANSLATOR_SET_ERROR;
  error: string;
}

export interface ITranslatorFlushTranslation {
  type: typeof TRANSLATOR_FLUSH_TRANSLATION;
}

export type TranslatorActionTypes =
  | ITranslatorSetMode
  | ITranslatorFetchMichelineToMichelson
  | ITranslatorFetchMichelsonToMicheline
  | ITranslatorSetMicheline
  | ITranslatorSetMichelson
  | ITranslatorSetError
  | ITranslatorFlushTranslation;
