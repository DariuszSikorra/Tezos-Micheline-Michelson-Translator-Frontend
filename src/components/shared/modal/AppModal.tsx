import React from "react";
import "./AppModal.scss";

const AppModal = ({ showModal, setShowModal, children }: any) =>
  showModal ? (
    <div className="app-modal" onClick={() => setShowModal(false)}>
      <div className="app-modal_content">
        <div className="app-modal_children" onClick={e => e.stopPropagation()}>
          {children}
        </div>
      </div>
    </div>
  ) : null;

export default AppModal;
