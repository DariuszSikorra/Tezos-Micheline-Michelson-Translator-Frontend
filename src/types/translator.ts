export type ModesType = 'michelinemichelson' | 'michelsonmicheline';

export const Labels = {
  michelinemichelson: 'Micheline to Michelson',
  michelsonmicheline: 'Michelson to Micheline'
};
