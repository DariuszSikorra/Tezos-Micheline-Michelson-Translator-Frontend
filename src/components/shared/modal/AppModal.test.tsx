import React from "react";
import { mount } from "enzyme";
import AppModal from "./AppModal";
import { AppModalProps } from "./types";

const defaultProps: AppModalProps = {
  showModal: true,
  setShowModal: () => {},
  children: <div>children</div>,
};

describe("AppModal", () => {
  const wrapper = mount(<AppModal {...defaultProps} />);

  it("matches snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const item = wrapper.find(AppModal);

    expect(
      item
        .at(0)
        .find(".app-modal_children")
        .at(0)
        .children()
        .at(0)
        .html(),
    ).toEqual("<div>children</div>");
  });
});
