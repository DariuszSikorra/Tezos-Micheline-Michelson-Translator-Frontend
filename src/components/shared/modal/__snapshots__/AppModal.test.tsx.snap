// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppModal matches snapshot 1`] = `
"<AppModal showModal={true} setShowModal={[Function: setShowModal]}>
  <div className=\\"app-modal\\" onClick={[Function: onClick]}>
    <div className=\\"app-modal_content\\">
      <div className=\\"app-modal_children\\" onClick={[Function: onClick]}>
        <div>
          children
        </div>
      </div>
    </div>
  </div>
</AppModal>"
`;
